### PR TITLE
api created and tested plus buyer dashboard updated

### PR DESCRIPTION
**Pull request by shreyashsing**

### CODE REVIEW

File: Backend/controllers/seller.controller.js [hunk 1]
```diff
@@ -351,6 +353,60 @@ const getAllDeals = asyncHandler(async (req, res) => {
...
+
+    // Validate input
+    if (!email || !phoneNumber) {
+        throw new ApiError(400, "Email and Phone Number are required.");
+    }
+
+    // Check if the seller exists based on email
+    let seller = await SellerDetails.findOne({ email });
+
+    if (seller) {
```
Comment (rishi2025) [inline Backend/controllers/seller.controller.js]: If seller is not found, just throw an error

File: Backend/controllers/seller.controller.js [hunk 2]
```diff
@@ -351,6 +353,60 @@ const getAllDeals = asyncHandler(async (req, res) => {
...
+        seller = new SellerDetails({
+            email,
+            phoneNumber,
+            address,
+            tags
+        });
+
+        // Save the new seller to the database
+        await seller.save();
+
```
Comment (rishi2025) [inline Backend/controllers/seller.controller.js]: Check if details are saved or not, then return the updated details.

File: Backend/controllers/seller.controller.js [hunk 3]
```diff
@@ -351,6 +353,60 @@ const getAllDeals = asyncHandler(async (req, res) => {
...
+        return res.status(201).json(
+            new ApiResponse(201, seller, "Seller information created successfully.")
+        );
+    }
+});
+
+const getTotalOrders = asyncHandler(async (req, res) => {
+    try {
+        const totalOrders = await PreviousDeals.countDocuments({});
```
Comment (rishi2025) [inline Backend/controllers/seller.controller.js]: total orders are to be counted for individual sellers

File: Backend/controllers/seller.controller.js [hunk 4]
```diff
@@ -351,6 +353,60 @@ const getAllDeals = asyncHandler(async (req, res) => {
...
+        return res.status(200).json(
+            new ApiResponse(200, { totalOrders }, "Total orders fetched successfully.")
+        );
+    } catch (error) {
+        console.error("Error fetching total orders:", error);
+        throw new ApiError(500, "Error fetching total orders.");
```
Comment (rishi2025) [inline Backend/controllers/seller.controller.js]: Make the error message -"Internal Server Error, An error occured while fetching total orders"

File: Backend/models/sellerDetails.model.js [hunk 5]
```diff
@@ -2,35 +2,32 @@ import mongoose, { Schema } from "mongoose";
 import mongooseAggregatePaginate from "mongoose-aggregate-paginate-v2";
 
 const sellerDetailsSchema = new Schema({
-
     email: {
-        type: Schema.Types.ObjectId,
-        ref: "Seller"
```
Comment (rishi2025) [inline Backend/models/sellerDetails.model.js:8]: Email is supposed to be a reference only.

File: src/components/BuyerDashboard.jsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+import { Container, RecentOrders, ProductRequest, Notification, SmallBox } from '../components/mainComponent.jsx';
+import { Graph, People } from '../Images/MainImage.jsx';
+
+function BuyerDashboard() {
+    return (
+        <Container className="min-h-[45.5rem] bg-[#F1EAFF] flex flex-col pt-10 gap-10" px="px-0">
+            <Container className="flex flex-row justify-between">
+                <SmallBox
+                    imagesrc={Graph}
+                    title={"Total Orders"}
```
Comment (rishi2025) [inline src/components/BuyerDashboard.jsx:14]: Props should be passed for these, with appropriate default values

File: src/components/LeftPanel.jsx [hunk 7]
```diff
@@ -27,7 +27,7 @@ const LeftPanel = ({
     }
 
     return (
-        <div className="w-[13rem] h-[45.5rem] p-5 left-0 bg-gray-100">
```
Comment (rishi2025) [inline src/components/LeftPanel.jsx:30]: Avoid changing the dimensions of finalized components.

**Review by rishi2025 — CHANGES_REQUESTED** [review]
Modifications requested

File: Backend/controllers/seller.controller.js (hunk 1 above)
Comment (shreyashsing) [reply Backend/controllers/seller.controller.js]: required modifications is done.

File: Backend/controllers/seller.controller.js (hunk 3 above)
Comment (shreyashsing) [reply Backend/controllers/seller.controller.js]: required modifications are done.

File: Backend/controllers/seller.controller.js (hunk 4 above)
Comment (shreyashsing) [reply Backend/controllers/seller.controller.js]: required modifications are done.

File: Backend/models/sellerDetails.model.js (hunk 5 above)
Comment (shreyashsing) [reply Backend/models/sellerDetails.model.js:8]: required modifications are done.

File: src/components/BuyerDashboard.jsx (hunk 6 above)
Comment (shreyashsing) [reply src/components/BuyerDashboard.jsx:14]: required modifications are done.

File: src/components/LeftPanel.jsx (hunk 7 above)
Comment (shreyashsing) [reply src/components/LeftPanel.jsx:30]: required modifications are done.

**Pull request merged**